plugins/modules/route53_zone.py

```diff
@@ -222,6 +320,68 @@ def find_zones(zone_in, private_zone):
     return zones
 
 
+def get_dnssec(hosted_zone_id: str) -> Dict[str, Any]:
+    try:
+        return client.get_dnssec(HostedZoneId=hosted_zone_id)
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg=f"Could not get dnssec details about hosted zone {hosted_zone_id}")
+
+
+def enable_hosted_zone_dnssec(zone_id: str) -> None:
+    try:
+        client.enable_hosted_zone_dnssec(HostedZoneId=zone_id)
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg=f"Could not enable DNSSEC for {zone_id}")
+
+
+def disable_hosted_zone_dnssec(zone_id: str) -> None:
+    try:
+        client.disable_hosted_zone_dnssec(HostedZoneId=zone_id)
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg=f"Could not enable DNSSEC for {zone_id}")
+
+
+def get_hosted_zone(hosted_zone_id: str) -> Dict[str, Any]:
+    try:
+        return client.get_hosted_zone(Id=hosted_zone_id)  # could be in different regions or have different VPCids
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg=f"Could not get details about hosted zone {hosted_zone_id}")
+
+
+def ensure_dnssec(zone_id: str) -> bool:
+    changed = False
+    dnssec = module.params.get("dnssec")
+
+    response = get_dnssec(zone_id)
+    dnssec_status = response["Status"]["ServeSignature"]
+
+    # If get_dnssec command output returns "NOT_SIGNING",
+    # the Domain Name System Security Extensions (DNSSEC) signing is not enabled for the
+    # Amazon Route 53 hosted zone.
+    if dnssec:
+        if dnssec_status == "NOT_SIGNING":
+            # Enable DNSSEC
+            if not module.check_mode:
+                enable_hosted_zone_dnssec(zone_id)
+            changed = True
+        elif dnssec_status == "DELETING":
+            # DNSSEC signing is in the process of being removed for the hosted zone.
+            module.warn(
+                f"DNSSEC signing is in the process of being removed for the hosted zone: {zone_id}."
+                "Could not enable it."
+            )
+    else:
+        if dnssec_status == "SIGNING":
+            # Disable DNSSEC
+            if not module.check_mode:
+                disable_hosted_zone_dnssec(zone_id)
+            changed = True
+        # if dnssec_status == "DELETING":
+        # DNSSEC signing is in the process of being removed for the hosted zone.
```

Comment:
```suggestion
```

Reply:
I added these comments to document what's happening if dnssec_status == "DELETING":.